In [12]:
import numpy as np
import pandas as pd
import statistics

### 1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6.  Cousin Oliver was 8.  Find mean, median, mode, variance, std dev, and std error of the ages.

In [13]:
#Generate data frame:
cast_age = pd.DataFrame()
cast_age['name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
cast_age['age'] = [14, 12, 11, 10, 8, 6, 8]

#Central tendency - mean, median, mode:
print('Mean = {}'.format(np.mean(cast_age['age'])))
print('Median = {}'.format(np.median(cast_age['age'])))
print('Mode = {}'.format(statistics.mode(cast_age['age'])))

#Measures of variance - variance, std deviation, std error:
print('Variance = {}'.format(cast_age['age'].var()))
print('Std Dev = {}'.format(np.std(cast_age['age'])))
print('Std Error = {}'.format((np.std(cast_age['age'], ddof=1))/(np.sqrt(len(cast_age['age'])))))


Mean = 9.857142857142858
Median = 10.0
Mode = 8
Variance = 7.476190476190475
Std Dev = 2.531435020952764
Std Error = 1.0334540197243192


### 2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

For this particular data set, I'd choose the median to best represent the central tendency.  The mode identifies the second youngest age as being the most common which biases the central tendancy low.  The mean is also quite low relative to all the data points because of Cindy's age.

I would choose the variance as the best indicator of variance in the dataset.  The variance is high, indicating there are datapoints that are considerably different from the mean.

### 3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [14]:
cast_age['name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
cast_age['age'] = [14, 12, 11, 10, 8, 7, 8]

#Central tendency - mean, median, mode:
print('Mean = {}'.format(np.mean(cast_age['age'])))
print('Median = {}'.format(np.median(cast_age['age'])))
print('Mode = {}'.format(statistics.mode(cast_age['age'])))

#Measures of variance - variance, std deviation, std error:
print('Variance = {}'.format(cast_age['age'].var()))
print('Std Dev = {}'.format(np.std(cast_age['age'])))
print('Std Error = {}'.format((np.std(cast_age['age'], ddof=1))/(np.sqrt(len(cast_age['age'])))))


Mean = 10.0
Median = 10.0
Mode = 8
Variance = 6.333333333333333
Std Dev = 2.32992949004287
Std Error = 0.9511897312113418


The mean, variance, std dev, and std error changed.  The mean increased while the std dev and std error decreased, indicating the individual datapoints are more similiar to the central tendency than they were before.  The median and mode did not change.

### 4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [11]:
cast_age['name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica']
cast_age['age'] = [14, 12, 11, 10, 8, 7, 1]

#Central tendency - mean, median, mode:
print('Mean = {}'.format(np.mean(cast_age['age'])))
print('Median = {}'.format(np.median(cast_age['age'])))

#mode
(values, counts) = np.unique(cast_age['age'], return_counts=True)
ind = np.argmax(counts)
print((values, counts))
values[ind]
print('Mode = {}'.format(values[ind]))

#Measures of variance - variance, std deviation, std error:
print('Variance = {}'.format(cast_age['age'].var()))
print('Std Dev = {}'.format(np.std(cast_age['age'])))
print('Std Error = {}'.format((np.std(cast_age['age'], ddof=1))/(np.sqrt(len(cast_age['age'])))))


Mean = 9.0
Median = 10.0
(array([ 1,  7,  8, 10, 11, 12, 14], dtype=int64), array([1, 1, 1, 1, 1, 1, 1]))
Mode = 1
Variance = 18.0
Std Dev = 3.927922024247863
Std Error = 1.6035674514745462


Replacing Oliver with Jessica changes all of the central tendency and variance values except the median.  The mean and median are similar estimates of the central tendency.  There is no mode as all values occur only once.  The variance is very high because there is a large spread in the data.  The std dev and std error also increased.

### 5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

### Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [20]:
fans = pd.DataFrame()
fans['magazine'] = ['tv guide', 'entertainment weekly', 'pop culture today', 'sciphi']
fans['value'] = [20, 23, 17, 5]

#Central tendency - mean, median
print('Mean = {}'.format(np.mean(fans['value'])))
print('Median = {}'.format(np.median(fans['value'])))

#Measures of variance - variance, std deviation, std error:
print('Variance = {}'.format(fans['value'].var()))
print('Std Dev = {}'.format(np.std(fans['value'])))
print('Std Error = {}'.format((np.std(fans['value'], ddof=1))/(np.sqrt(len(fans['value'])))))


Mean = 16.25
Median = 18.5
Variance = 62.25
Std Dev = 6.832825184358224
Std Error = 3.944933459514875


Based on the answers, I would rely on the mean and estimate 16% of adult Americans were fans of the show on the 50th Anniversary.  There is large variance in the data, so using the mean is the best measure of central tendency because it incorporates all values.  